scrapy/pipelines/files.py

```diff
@@ -65,8 +71,8 @@ def stat_file(self, path: str, info):
 
         return {"last_modified": last_modified, "checksum": checksum}
 
-    def _get_filesystem_path(self, path: str) -> Path:
-        path_comps = path.split("/")
+    def _get_filesystem_path(self, path: Union[str, PathLike]) -> Path:
+        path_comps = _to_string(path).split("/")
```

Comment:
Type-hint wise, I think this change also implies that code calling this function (e.g. the 2 methods above) now would also accept `Union[str, PathLike]`, so we should update their type hints accordingly.

Reply:
Fixed. thank you.